testnet-contracts/contracts/CosmosBridge.sol

```diff
@@ -0,0 +1,295 @@
+        Null,
+        Pending,
+        Success,
+        Failed
+    }
+
+    struct BridgeClaim {
+        uint256 nonce;
+        bytes cosmosSender;
+        address payable ethereumReceiver;
+        address originalValidator;
+        address tokenAddress;
+        string symbol;
+        uint256 amount;
+        Status status;
+    }
+
+    /*
+    * @dev: Event declarations
+    */
+    event LogOracleSet(
+        address _oracle
+    );
+
+    event LogBridgeBankSet(
+        address _bridgeBank
+    );
+
+    event LogNewBridgeClaim(
+        uint256 _bridgeClaimCount,
+        uint256 _nonce,
+        bytes _cosmosSender,
+        address payable _ethereumReceiver,
+        address _validatorAddress,
+        address _tokenAddress,
+        string _symbol,
+        uint256 _amount
+    );
+
+    event LogBridgeClaimCompleted(
+        uint256 _bridgeClaimID
+    );
+
+    /*
+    * @dev: Modifier to restrict access to completed BridgeClaims
+    */
+    modifier isPending(
+        uint256 _bridgeClaimID
+    )
+    {
+        require(
+            isBridgeClaimActive(_bridgeClaimID),
+            "Bridge claim is not active"
+        );
+        _;
+    }
+
+    /*
+    * @dev: Modifier to restrict access to the operator.
+    */
+    modifier onlyOperator()
+    {
+        require(
+            msg.sender == operator,
+            'Must be the operator.'
+        );
+        _;
+    }
+
+      /*
+    * @dev: The bridge is not active until oracle and bridge bank are set
+    */
+    modifier isActive()
+    {
+        require(
+            hasOracle == true && hasBridgeBank == true,
+            "The Operator must set the oracle and bridge bank for bridge activation"
+        );
+        _;
+    }
+
+    /*
+    * @dev: Constructor
+    */
+    constructor(
+        address _operator,
+        address _valset
+    )
+        public
+    {
+        bridgeClaimCount = 0;
+        operator = _operator;
+        valset = Valset(_valset);
+        hasOracle = false;
+        hasBridgeBank = false;
+    }
+
+    /*
+    * @dev: setOracle
+    */
+    function setOracle(
+        address _oracle
+    )
+        public
+        onlyOperator
+    {
+        require(
+            !hasOracle,
+            "The Oracle cannot be updated once it has been set"
+        );
+
+        hasOracle = true;
+        oracle = _oracle;
+
+        emit LogOracleSet(
+            oracle
+        );
+    }
+
+    /*
+    * @dev: setBridgeBank
+    */
+    function setBridgeBank(
+        address payable _bridgeBank
+    )
+        public
+        onlyOperator
+    {
+        require(
+            !hasBridgeBank,
+            "The Bridge Bank cannot be updated once it has been set"
+        );
+
+        hasBridgeBank = true;
+        bridgeBank = BridgeBank(_bridgeBank);
+
+        emit LogBridgeBankSet(
+            address(bridgeBank)
+        );
+    }
+
+    // TODO: BridgeClaims can only be created for BridgeTokens on BridgeBank's whitelist.
```

Comment:
Remove TODO, make this a normal comment. We probably need to document this whitelisting step in the README more clearly too which can be done in follow up PR. The current implementation will then result in unfinalized prophecies left dangling until the token has been whitelisted, which I think could be the right choice as it will allow easy checks for this situation, allowing the operator to remedy it as needed when it comes up. <-- probably want to add these comments into the code comments or README too.

Reply:
https://github.com/cosmos/peggy/pull/82/commits/f34f19bb8544b4e8a153af2152647283ffa8be13